### PR TITLE
Add support for referenced endpoints object

### PR DESCRIPTION
**Pull request by AmaliMatharaarachchi**
### Issues

Fixes https://github.com/wso2/product-microgateway/issues/2344

### Automation tests
 - Unit tests added: Yes
 - Integration tests added: Yes - but cannot enable due to this issue https://github.com/wso2/product-apim-tooling/issues/821

### Tested environments

Tested on catalina

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/internal/oasparser/envoyconf/routes_with_clusters.go [hunk 1]
```diff
@@ -88,15 +88,30 @@ func CreateRoutesWithClusters(mgwSwagger model.MgwSwagger, upstreamCerts []byte,
...
+			endpoints = append(endpoints, addresses...)
+		}
+	}
+
 	// check API level production endpoints available
 	if mgwSwagger.GetProdEndpoints() != nil && len(mgwSwagger.GetProdEndpoints().Endpoints) > 0 {
 		apiLevelEndpointProd = mgwSwagger.GetProdEndpoints()
-		apiLevelClusterNameProd = strings.TrimSpace(organizationID + "_" + prodClustersConfigNamePrefix + vHost + "_" +
+		apiLevelClusterNameProd = strings.TrimSpace(organizationID + "_" + apiLevelEndpointProd.EndpointPrefix + "_" + vHost + "_" +
 			strings.Replace(mgwSwagger.GetTitle(), " ", "", -1) + mgwSwagger.GetVersion())
```
Comment (renuka-fernando) [inline adapter/internal/oasparser/envoyconf/routes_with_clusters.go:108]: what if we have one place to generate cluster name, with the expected prefix

Comment (AmaliMatharaarachchi) [reply adapter/internal/oasparser/envoyconf/routes_with_clusters.go]: fixed

File: adapter/internal/oasparser/envoyconf/routes_with_clusters.go [hunk 2]
```diff
@@ -68,238 +69,238 @@ func CreateRoutesWithClusters(mgwSwagger model.MgwSwagger, upstreamCerts []byte,
...
 	}
 	return routes, clusters, endpoints
 }
 
+func getClusterName(epPrefix string, organizationID string, vHost string, swaggerTitle string, swaggerVersion string,
+	resourceID string) string {
+	if resourceID != "" {
+		return strings.TrimSpace(organizationID+"_"+epPrefix+"_"+vHost+"_"+strings.Replace(swaggerTitle, " ", "", -1)+swaggerVersion) +
+			"_" + strings.Replace(resourceID, " ", "", -1) + "0"
+	}
```
Comment (pubudu538) [inline adapter/internal/oasparser/envoyconf/routes_with_clusters.go:288]: Formatting issues.

**Review by pubudu538 — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[succeeded] dev-deployment : 20211202.25

**Pull request merged**